asserts/database.go

```diff
@@ -157,13 +157,17 @@ func (db *Database) GenerateKey(authorityID string) (fingerprint string, err err
 		return "", fmt.Errorf("failed to generate private key: %v", err)
 	}
 
-	return db.keypairMgr.Import(authorityID, OpenPGPPrivateKey(privKey))
+	return db.ImportKey(authorityID, OpenPGPPrivateKey(privKey))
 }
 
 // ImportKey stores the given private/public key pair for identity and
-// returns its fingerprint
-func (db *Database) ImportKey(authorityID string, privKey PrivateKey) (fingerprint string, err error) {
-	return db.keypairMgr.Import(authorityID, privKey)
+// returns its key id.
+func (db *Database) ImportKey(authorityID string, privKey PrivateKey) (keyID string, err error) {
```

Comment:
@niemeyer given that you are overall happy I can proceed with that, it needs some further mechanical test cleanups